executor/builder.go

```diff
@@ -1808,7 +1808,16 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 	e.innerCtx.keyCols = innerKeyCols
 	e.joinResult = newFirstChunk(e)
 	executorCounterIndexLookUpJoin.Inc()
-	return e
+	idxHash := &IndexNestedLoopHashJoin{
```

Comment:
Why do we wrap an `IndexHashJoin` here?
Please add some comments.

Reply:
To promise all the IndexJoin can execute the logic of IndexHashJoin.
This wrap will be removed before this PR is merged.